core/sources/utf8_utils.move

```diff
@@ -12,6 +12,36 @@ module aptos_names::utf8_utils {
     const CHARACTER_SET_LATIN_LOWER_ALPHA_IDX: u64 = 0;
     const CHARACTER_SET_LATIN_DIGIT_IDX: u64 = 1;
 
+    const MOVE_DOMAIN_SUFFIX: vector<u8> = b".move";
+
+    public fun ends_with_str(str: String, suffix_bytes: &vector<u8>): bool {
+        let u8_str = string::bytes(&str);
+        let u64_str = utf8_to_vec_u64_fast(u8_str);
+        ends_with(&u64_str, suffix_bytes)
+    }
+
+    // Returns |true| if |str| ends with |suffix_bytes|.
+    public fun ends_with(str: &vector<u64>, suffix_bytes: &vector<u8>): bool {
+        let suffix = utf8_to_vec_u64_fast(suffix_bytes);
+        let suffix_len = vector::length(&suffix);
+        let str_len = vector::length(str);
+        if (suffix_len > str_len) {
```

Comment:
Is there a reason that we are not using `>=`? as an example, I feel like maybe `.move` is not a viable domain name either?

Reply:
ends_with should return true for calls like `ends_with(".xyz", ".xyz")` because ".xzy" ends with ".xyz"